### PR TITLE
[ARCTIC-1646][Bug]: Build failed when change hadoop version to 3.2.1

### PR DESCRIPTION
**Pull request by zhaomin1423**
## Why are the changes needed?

close https://github.com/NetEase/arctic/issues/1646#issue-1787080167
## Brief change log

update org.apache.commons.lang.* to org.apache.commons.lang3.* in AMS module.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhoujinsong** [comment]
Thanks a lot for your contribution!

Based on your code, I was able to build successfully, but when starting AMS, the following error occurred:
351687333241_ pic

It seems to be caused by a conflict in the Guava dependency. Can you also fix this issue?

**Comment by zhaomin1423** [comment]
> Thanks a lot for your contribution!
> 
> Based on your code, I was able to build successfully, but when starting AMS, the following error occurred: 351687333241_ pic
> 
> It seems to be caused by a conflict in the Guava dependency. Can you also fix this issue?

Guava is used by hadoop common, hive-metastore and this project, Can we explicitly declare a guava dependency？

**Comment by zhoujinsong** [comment]
> Can we explicitly declare a guava dependency？

Yes, although the Arctic project depends on the Guava package that the Iceberg project shade on, in order to solve the conflict between Hadoop and Hive's dependencies on Guava, specifying the Guava version that the project depends on may be a good solution.

However, we may still need to validate this solution by conducting actual deployment testing to ensure that it does not introduce any other issues.

**Comment by zhaomin1423** [comment]
> > Can we explicitly declare a guava dependency？
> 
> Yes, although the Arctic project depends on the Guava package that the Iceberg project shade on, in order to solve the conflict between Hadoop and Hive's dependencies on Guava, specifying the Guava version that the project depends on may be a good solution.
> 
> However, we may still need to validate this solution by conducting actual deployment testing to ensure that it does not introduce any other issues.

Thanks, I build and start ams to do few simple tests.

File: ams/server/src/test/java/com/netease/arctic/server/optimizing/flow/view/MatchResult.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public boolean isOk() {
 
   @Override
   public String toString() {
-    return Objects.toStringHelper(this)
+    return MoreObjects.toStringHelper(this)
```
Comment (zhoujinsong) [inline ams/server/src/test/java/com/netease/arctic/server/optimizing/flow/view/MatchResult.java:63]: In the Arctic project, I suggest using the version of Guava Shaded by the Iceberg project as a dependency. Therefore, the imported classes here can be changed to: `org.apache.iceberg.relocated.com.google.common.base.MoreObjects`

File: pom.xml [hunk 2]
```diff
@@ -341,6 +342,42 @@
                 <version>${awssdk.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>com.google.guava</groupId>
+                <artifactId>guava</artifactId>
+                <version>${guava.version}</version>
+                <exclusions>
```
Comment (zhoujinsong) [inline pom.xml:349]: Is there any reason or basis for the content excluded here?

File: core/pom.xml [hunk 3]
```diff
@@ -116,6 +116,11 @@
             <classifier>nohive</classifier>
         </dependency>
 
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
```
Comment (zhoujinsong) [inline core/pom.xml:122]: Now the Flink/Trino module also has its own declared Guava dependency. If the dependency is declared in the core, it seems that the dependencies of other modules can be deleted.

**Review by zhoujinsong — COMMENTED** [review]
@zhaomin1423 
Here is the code I previously attempted to solve this problem, which can be used as a reference: https://github.com/NetEase/arctic/compare/master...zhoujinsong:arctic:adapt-commons-dependency

**Comment by zhaomin1423** [comment]
> @zhaomin1423 Here is the code I previously attempted to solve this problem, which can be used as a reference: [master...zhoujinsong:arctic:adapt-commons-dependency](https://github.com/NetEase/arctic/compare/master...zhoujinsong:arctic:adapt-commons-dependency)

Maybe adding a hadoop-3 profile to declare hadoop-client-api as dependency is more suitable for hadoop3.

**Comment by zhoujinsong** [comment]
fixed by 1728.

**Pull request closed without merging**